cranelift-codegen/src/binemit/relaxation.rs

```diff
@@ -135,7 +136,11 @@ pub fn relax_branches(
     let jumptables_size = offset - jumptables;
     let rodata = offset;
 
-    // TODO: Once we have constant pools we'll do some processing here to update offset.
+    let constants: Vec<Constant> = func.dfg.constants.iter().map(|(h, _)| h.clone()).collect();
+    for constant_handle in constants {
+        func.dfg.constants.set_offset(constant_handle, offset);
+        offset += func.dfg.constants.get(constant_handle).len() as u32
```

Comment:
We should use `try_from()` to convert to `u32` here. It may not be possible to create a 4 GiB constant today, but if anyone ever could, silent wraparound here would be a subtle bug.